core/src/main/scala/com/github/gpoirier/barrage/actions.scala

```diff
@@ -60,11 +61,20 @@ object actions {
       object Three extends Workshop(3)
     }
 
-    sealed trait MachineShop extends Action
+    case class MachineShop(machinery: Machinery) extends Action {
```

Comment:
I shouldn't have made Workshop a case class to create case objects of it to pattern match on, the match isn't guaranteed to be exhaustive. It would be better to make this a sealed abstract class and add val in front of `machinery: Machinery` so that the field is still accessible outside of this class. Or possibly instead add a reward field instead (defined abstractly on `Action` probably) as mentioned on the other comment.